src/session/get-access-token.ts

```diff
@@ -114,16 +114,11 @@ export default function accessTokenFactory(
 
       // Update the session.
       const newSession = fromTokenSet(tokenSet, config);
-      sessionCache.set(
-        req,
-        res,
-        fromJson({
-          ...session,
-          ...newSession,
-          refreshToken: newSession.refreshToken || session.refreshToken,
-          user: { ...session.user, ...newSession.user }
-        })
-      );
+      Object.assign(session, {
+        ...newSession,
+        refreshToken: newSession.refreshToken || session.refreshToken,
+        user: { ...session.user, ...newSession.user }
+      });
```

Comment:
Does this get saved in the session cache?

Reply:
Yep - the session is mutable and whatever is in it at the end of the request chain [will get persisted ](https://github.com/auth0/nextjs-auth0/blob/main/src/session/cache.ts#L22)

Reply:
The second request to `/api/session` in the test should also verify this https://github.com/auth0/nextjs-auth0/pull/307/files#diff-43549841cd306e45cf537adad769f7fc683abaf46a9eab1146d958238493af2aR112-R113